lcm/test/lcm_trajectory_test.cc

```diff
@@ -0,0 +1,122 @@
+static const char DATATYPE_1[] = "DATATYPE_1";
+static const char DATATYPE_2[] = "DATATYPE_2";
+static const int NUM_DATAPOINTS = 5;
+static const int NUM_DATATYPES = 2;
+static const int NUM_TRAJECTORIES = 2;
+
+
+class LcmTrajectoryTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    datatypes_.resize(NUM_DATAPOINTS);
+    datatypes_[0] = DATATYPE_1;
+    datatypes_[1] = DATATYPE_2;
+
+    traj_1_.traj_name = TEST_TRAJ_NAME_1;
+    traj_1_.time_vector = VectorXd::Ones(NUM_DATAPOINTS);
+    traj_1_.datapoints = MatrixXd::Identity(NUM_DATAPOINTS, NUM_DATAPOINTS);
+    traj_1_.datatypes = datatypes_;
+
+    traj_2_.traj_name = TEST_TRAJ_NAME_2;
+    traj_2_.time_vector = VectorXd::Zero(NUM_DATAPOINTS);
+    traj_2_.datapoints = MatrixXd::Identity(NUM_DATAPOINTS, NUM_DATAPOINTS);
+    traj_2_.datatypes = datatypes_;
+
+    metadata_.datetime = "";
+    metadata_.git_dirty_flag = false;
+    metadata_.name = TEST_NAME;
+    metadata_.description = TEST_DESCRIPTION;
+
+    trajectory_names_.resize(NUM_TRAJECTORIES);
+    trajectory_names_[0] = TEST_TRAJ_NAME_1;
+    trajectory_names_[1] = TEST_TRAJ_NAME_2;
+    trajectories_.resize(NUM_TRAJECTORIES);
+    trajectories_[0] = traj_1_;
+    trajectories_[1] = traj_2_;
+
+    lcm_traj_ = LcmTrajectory(trajectories_,
+                              trajectory_names_,
+                              TEST_NAME,
+                              TEST_DESCRIPTION);
+  }
+
+  lcmt_trajectory_block traj_block_1_;
+  lcmt_trajectory_block traj_block_2_;
+  lcmt_saved_traj lcmt_traj_;
+  LcmTrajectory::Trajectory traj_1_;
+  LcmTrajectory::Trajectory traj_2_;
+  vector<string> datatypes_;
+  vector<LcmTrajectory::Trajectory> trajectories_;
+  vector<string> trajectory_names_;
+  lcmt_metadata metadata_;
+  LcmTrajectory lcm_traj_;
+};
+
+TEST_F(LcmTrajectoryTest, TestSavingAndLoadingTrajectory) {
+  lcm_traj_.writeToFile(TEST_FILEPATH);
+
+  lcmt_saved_traj loaded_lcm_traj = lcm_traj_.loadFromFile(TEST_FILEPATH);
+
+  EXPECT_EQ(loaded_lcm_traj.num_trajectories, NUM_TRAJECTORIES);
+  EXPECT_EQ(loaded_lcm_traj.trajectory_names.size(), NUM_TRAJECTORIES);
+  EXPECT_EQ(loaded_lcm_traj.trajectories.size(), NUM_TRAJECTORIES);
+  EXPECT_EQ(loaded_lcm_traj.metadata.datetime, metadata_.datetime);
+  EXPECT_EQ(loaded_lcm_traj.metadata.name, TEST_NAME);
+  EXPECT_EQ(loaded_lcm_traj.metadata.description, TEST_DESCRIPTION);
+  EXPECT_EQ(loaded_lcm_traj.metadata.git_dirty_flag, metadata_.git_dirty_flag);
+}
+
+TEST_F(LcmTrajectoryTest, TestConstructorFromLcmTObject) {
+  lcm_traj_.writeToFile(TEST_FILEPATH);
+
+  LcmTrajectory loaded_traj = LcmTrajectory(lcm_traj_.loadFromFile(
+                                TEST_FILEPATH));
+  EXPECT_TRUE(
+    loaded_traj.trajectories_[TEST_TRAJ_NAME_1].time_vector.isApprox(
+      lcm_traj_.trajectories_[TEST_TRAJ_NAME_1].time_vector));
+  EXPECT_TRUE(
+    loaded_traj.trajectories_[TEST_TRAJ_NAME_1].time_vector.isApprox(
```

Comment:
Looks like a duplicate check on `TEST_TRAJ_NAME_1`. Should this be `TEST_TRAJ_NAME_2`?